### PR TITLE
reuse moduleidre

### PR DESCRIPTION
**Pull request by iambumblehead**
* reuse one regexp rather than create separate regexps
 * remove esmockIsLoader.js file, export that from esmockLoader.js instead

### CODE REVIEW

File: src/esmockLoader.js [hunk 1]
```diff
@@ -121,6 +122,7 @@ const resolve = async (specifier, context, nextResolve) => {
 const loaderVerificationQuery = '?esmock-loader=true'
 const loaderIsVerified = url => import(
   url + loaderVerificationQuery).then(m => m.default === true)
+const loaderIs = (c => () => (c = c || loaderIsVerified(import.meta.url)))()
```
Comment (koshic) [inline src/esmockLoader.js:125]: 1. Loader is what? Let's stop saving bytes on variable names ) Sure, for such small project we can even use 'a', 'b', and so on, but why? Terser or other minifier do the same, if we really want. 
2. Do we still need 'loaderIsVerified' / 'loaderIs' pair? Looks like 'loaderIsVerified' never exported, and can be combined with 'loaderIs'.

**Review by koshic — CHANGES_REQUESTED** [review]

File: src/esmockLoader.js (hunk 1 above)
Comment (iambumblehead) [reply src/esmockLoader.js:125]: thanks I will update tomorrow because its 3am here now

Comment (iambumblehead) [reply src/esmockLoader.js]: @koshic this is what it looks like after incorporating your advice. What do you think?
```javascript
const loaderVerificationUrl = urlDummy + '?esmock-loader=true'
const loaderIsVerified = (memo => () => memo = memo || (
  import(loaderVerificationUrl).then(m => m.default === true)))()
const load = async (url, context, nextLoad) => {
  if (url === loaderVerificationUrl) {
    return {
      format: 'module',
      shortCircuit: true,
      responseURL: url,
      source: 'export default true'
    }
  }
// ...
```

Comment (koshic) [reply src/esmockLoader.js]: Looks better, thx!

But looks like there is an issue, not related to that PR. I mean return type of 'loaderIsVerified' fn. Let's try almost the same code:
```js
import { setTimeout } from "node:timers/promises";

const loaderIsVerified = (
  (memo) => () =>
    (memo = memo || setTimeout(5000).then(() => true))
)();

const start = Date.now();

const VALUE1 = loaderIsVerified();
console.log(VALUE1, VALUE1 === true, Date.now() - start);

await setTimeout(7000);

const VALUE2 = loaderIsVerified();
console.log(VALUE2, VALUE2 === true, Date.now() - start);

const VALUE3 = await loaderIsVerified();
console.log(VALUE3, VALUE3 === true, Date.now() - start);
```

image

As expected, without await we will receive Promise instance: VALUE1 is pending, and VALUE2 is resolved. Sadly, we can't compare those values to string (which we want to do in 'if (url === loaderVerificationUrl) {'). Only VALUE3 contains boolean.

Comment (iambumblehead) [reply src/esmockLoader.js]: The only caller of this function uses the await keyword, as the previous verification function was called.

Politely, I don't see a problem

Comment (koshic) [reply src/esmockLoader.js]: Yeah, missed
```js
  if (!await esmockIsLoader())
    throw esmockErr.errMissingLoader()
```

So - approved!

**Review by koshic — APPROVED** [review]

File: src/esmockLoader.js (hunk 1 above)
Comment (iambumblehead) [reply src/esmockLoader.js]: awesome!

Comment (iambumblehead) [reply src/esmockLoader.js]: thanks for the suggestions and review

Comment (koshic) [reply src/esmockLoader.js]: > What do you recommend?
@iambumblehead PR is closed, but... )

Personally I prefer something like that (if we still want to use 'less code / one-line expressions' pattern):
```js
let loaderIsVerifiedCache;
const loaderIsVerified = async () =>
  (loaderIsVerifiedCache ??= (await import(importUrl)).default === true);
```

Why? It's much more clear (fewer brackets / functions), no mixed syntax (async/await and than/catch).

Comment (iambumblehead) [reply src/esmockLoader.js]: I think I agree. One thing, however, the "??=" syntax isn't supported in node 14 https://node.green/#ES2021-features-Logical-Assignment-----basic-support

There was a problem with the node 14 test ci, not related to esmock, so the pipeline was removed BUT esmock does work with node 14 still (and possibly node 12 still). Perhaps we should completely drop node 14 and support latest node 16 only.

Comment (koshic) [reply src/esmockLoader.js]: I think there are no reasons to support Node 14 due to
https://github.com/nodejs/release#release-schedule
image

Moreover, Node 16 will reach EOL this September. Good opportunity to release esmock 3.0.0 without legacy support )

Comment (iambumblehead) [reply src/esmockLoader.js]: @koshic https://github.com/iambumblehead/esmock/pull/232

**Pull request merged**